src/main/java/reader/DataReader.java

```diff
@@ -81,8 +82,8 @@ public String readSchema(String schemaPath) {
     return schema.toString();
   }
   // write json file
-  public void writeFile(String json) {
-    try (FileWriter myWriter = new FileWriter("route.json")) {
+  public void writeFile(String json, String path) {
```

Comment:
please change the variable name "json" to something more generic like fileContent, I think you might want to use this method in the future to save different types of files not only json